src/Model.php

```diff
@@ -419,4 +446,258 @@ private function setDefaultTable()
+	 *
+	 * @param   string  $key attribute name
+	 * @return void
+	 */
+	public function __unset($key)
+	{
+		$this->offsetUnset($key);
+	}
+
+	/**
+	 * Convert the model instance to an array.
+	 *
+	 * @return array
+	 */
+	public function toArray()
+	{
+
+		// TODO relations
+		return array_merge($this->getAttributes());
+
+		// A return array_merge($this->getAttributes(), $this->getRelations());
+	}
+
+	/**
+	 * Convert the model instance to JSON.
+	 *
+	 * @param   int  $options ?
+	 * @return string
+	 *
+	 * @throws JsonEncodingException
+	 */
+	public function toJson($options = 0)
+	{
+		$json = json_encode($this->jsonSerialize(), $options);
+
+		if (JSON_ERROR_NONE !== json_last_error())
+		{
+			throw JsonEncodingException::forModel($this, json_last_error_msg());
+		}
+
+		return $json;
+	}
+
+	/**
+	 * Convert the object into something JSON serializable.
+	 *
+	 * @return array
+	 */
+	public function jsonSerialize()
+	{
+		return $this->toArray();
+	}
+
+	/**
+	 * Increment a column's value by a given amount.
+	 *
+	 * @param   string     $column ?
+	 * @param   float|int  $amount ?
+	 * @param   boolean    $lazy   laxy increment if true
+	 * @return integer
+	 */
+	protected function increment($column, $amount = 1, $lazy = false)
+	{
+		return $this->incrementOrDecrement($column, $amount, $lazy, 'increment');
+	}
+
+	/**
+	 * Decrement a column's value by a given amount.
+	 *
+	 * @param   string     $column ?
+	 * @param   float|int  $amount ?
+	 * @param   boolean    $lazy   laxy increment if true
+	 * @return integer
+	 */
+	protected function decrement($column, $amount = 1, $lazy = false)
+	{
+		return $this->incrementOrDecrement($column, $amount, $lazy, 'decrement');
+	}
+
+	/**
+	 * Run the increment or decrement method on the model.
+	 *
+	 * @param   string     $column ?
+	 * @param   float|int  $amount ?
+	 * @param   float|int  $lazy   ?
+	 * @param   string     $method ?
+	 * @return integer|Model
+	 */
+	protected function incrementOrDecrement($column, $amount, $lazy, $method)
+	{
+
+		$amount = $method == 'increment' ? $amount : $amount * -1;
+
+		$amount = $amount + $this->$column;
+
+		$this->setAttribute($column, $amount);
+
+		if ($lazy)
+		{
+			return $this;
+		}
+
+		if ($this->exists)
+		{
+			return $this->update();
+		}
+
+		return $this->save();
+	}
+
+	/**
+	 * Increment the underlying attribute value and sync with original.
+	 *
+	 * @param   string     $column ?
+	 * @param   float|int  $amount ?
+	 * @param   string     $method ?
+	 * @return void
+	 */
+	protected function incrementOrDecrementAttributeValue($column, $amount, $method)
+	{
+		$this->{$column} = $this->{$column} + ($method == 'increment' ? $amount : $amount * -1);
+
+		$this->syncOriginalAttribute($column);
+	}
+
+	/**
+	 * Adds the data set to be loaded into the database during setup
+	 *
+	 * @param   CsvDataSet $dataSet csv dataset for testing
+	 * @return CsvDataSet
+	 */
+	public static function addDataSet(CsvDataSet $dataSet)
```

Comment:
No no no! The dataset belongs in the testing class not in the main one shipped in the libraries